# Test hypotezy

 V tomto notebooku sa snazime s $95$% istotou dokazat nasledujucu hypotezu: 
   ### "Pokud viník nehody byl pod silným vlivem alkoholu, došlo častěji k těžkým zdravotním následkům."
Využijeme k tomu **$\chi^2$ chí-kvadrát test** (chi-square test).

Nacitame potrebne kniznice.

In [1]:
import pandas as pd
import scipy.stats

Nacitame si dataframe zo suboru a ponechame si iba stlpce, s ktorymi budeme narabat.

In [2]:
df = pd.read_pickle("accidents.pkl.gz")
df = df[["p11", "p13a", "p13b"]]
df

p11  p13a  p13b
0         2     0     0
1         2     0     0
2         2     0     0
3         9     0     0
4         0     0     0
...     ...   ...   ...
487156    0     0     0
487157    2     0     0
487158    0     0     0
487159    2     0     0
487160    0     0     0

[487161 rows x 3 columns]

Z DataFramu odstranime nehody, kde bol vodic pod vplyvom drog.

In [3]:
df2 = df[(df['p11'] != 4) & (df['p11'] != 5)]
df2

p11  p13a  p13b
0         2     0     0
1         2     0     0
2         2     0     0
3         9     0     0
4         0     0     0
...     ...   ...   ...
487156    0     0     0
487157    2     0     0
487158    0     0     0
487159    2     0     0
487160    0     0     0

[485683 rows x 3 columns]

Kedze pracujeme s **$\chi^2$ testom** potrebujeme mat dva stlpce kategorickych dat, na ktorych budeme zistovat ich korelaciu. 

Pridame dva stlpce s boolovou hotnotou, reprezentujuce ci bol vodic pod plyvom alkoholu a ci doslo pri nehode k zraneniu alebo smrti.

In [4]:
df2["drunk"] = (df2['p11'] >= 7)
df2["injured"] = (df2['p13a'] + df2['p13b'] > 0)
df2

p11  p13a  p13b  drunk  injured
0         2     0     0  False    False
1         2     0     0  False    False
2         2     0     0  False    False
3         9     0     0   True    False
4         0     0     0  False    False
...     ...   ...   ...    ...      ...
487156    0     0     0  False    False
487157    2     0     0  False    False
487158    0     0     0  False    False
487159    2     0     0  False    False
487160    0     0     0  False    False

[485683 rows x 5 columns]

Vytvorime si kontingencnu tabulku pre vsetky styri kombinacie vstupov.

In [5]:
ct = pd.crosstab(df2['drunk'], df2['injured'])
ct

injured   False  True 
drunk                 
False    457528  10777
True      16492    886

Pomocou kontingencnej tabulky si vypocitame vsetky dolezite hodnoty. Ako napriklad p_value a chí-kvadrát statistiku. Tieto hodnoty sa daju peken vypocitat podla vzorcov, no existuje na to funckia z kniznice `scipy`, ktora nam poskytuje vsetky potrebne vysledne hodnoty.

In [6]:
scipy.stats.chi2_contingency(ct)

(558.1749514234125,
 2.0971505700338304e-123,
 1,
 array([[4.57059308e+05, 1.12456916e+04],
        [1.69606916e+04, 4.17308438e+02]]))

Hodnota p_value $2.09\cdot10^{-123}$ je vyrazne mensia než $0.05$. Proto zamietame nulovu hypotézu a mozeme jednoznacne prehlasit, že sa tu vyskytuje silna statisticky významná korelace mezi pomerom vodicov s vysokym alkoholom v krvi a zavaznym zraneniam alebo usmrteniu pri nehodach nimi  zapricinenych. 


Aj ked je pravdou ze korelace dvoch premennych neznamena aj kauzalitu. V tomto pripade sa vyskytuje velmi silne statisticky vyznamna korelace a  selsky rozum nam hovori, ze je to aj pripad kauzaity. 